src/parser/htmlParser.ts

```diff
@@ -8,13 +8,27 @@ import { findFirst } from '../utils/arrays';
 import { TokenType } from '../htmlLanguageTypes';
 import { isVoidElement } from '../languageFacts/fact';
 
+export interface AttributeWithLocation {
+	attr: string;
+	start: number;
+	end: number;
+	value: AttributeValueWithLocation | null;
```

Comment:
HTML documents can be very large, and so the number of nodes in a HTML AST.
When designing the AST, the focus was on keeping the size of each node as small as possible. That's why it doesn't keep all the offsets and originally also not the value. The idea is to rescan when that information is required.
That we keep the value should be reconsidered as for the workspace symbols we don't really need them in the AST (again, very easy to do with the scanner)